### ISSUE TITLE
Java: Issue with static initializer if no space after static

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#9417]
**Describe the bug**
Processing Java code
```
public class Test
{
  String a = "a", b = "b";
  static{
          a.trim();
          b.trim();
  }
}
```
results in the following xml snippet:
```
      <memberdef kind="function" id="classTest_1a09be701fca1d59a280713aa0f777506e" prot="package" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>b.</type>
        <definition>b. Test.trim</definition>
        <argsstring>()</argsstring>
        <name>trim</name>
        <qualifiedname>Test.trim</qualifiedname>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Test.java" line="6" column="12"/>
      </memberdef>
```
Obviously, ``trim`` is not a member. The issue does not occur if there is a space between ``static{`` or if there is only a single line in the initializer. Not posted here, in addition the initializer in xml consists only of the first line, rather than the whole code block.

**Expected behavior**
I expect that the resulting xml would contain no member named ``trim`` and that the initializer contained the whole code block, not just the first line. The expected xml can be produced by changing the source code to ``static {`` (i.e. with space).

**To Reproduce**
Find a Java file and doxygen config attached. The issue is demonstrated like mentioned above.
After running doxygen on the example, the files in the subfolder ``xml`` contain the snippet shown above.
[issue_no_space_after_static.zip](https://github.com/doxygen/doxygen/files/8978504/issue_no_space_after_static.zip)

**Version**
Tested with doxygen release 1.9.3.1:
```
/blah/doxygen/build/bin/doxygen -v
1.9.3.1
```
Unfortunately, I am currently not able to build newer doxygen versions.

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
TBH I did not check for duplicates. Please excuse if this is a known issue.
Also, as mentioned above, I am currently not able to build recent doxygen versions. Please excuse if the issue is already fixed.

**Comment by albert-github** [comment]
Disclaimer: I didn't test

Would the issue as reported in #8434 and the proposed pull request #8435 also fix this issue?

**Comment by Stewori** [comment]
I don't think so. I am using a doxygen fork with #8435 applied and still see this issue.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9431

Note: there might be a problem with the handling of the `static` as I see:

![image](https://user-images.githubusercontent.com/5223533/176430783-7b4ae1db-fd19-4b3c-8567-1e81670916d8.png)


and

![image](https://user-images.githubusercontent.com/5223533/176430864-80bf203b-5165-4ed5-8fc1-a03e8aa9a271.png)


What would you expect for the `static initializer` in this case? (best to create a separate issue out of it where the space is present so `static {`).

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
